### PR TITLE
alluxio: change default configuration in alluxio-site.properties

### PR DESCRIPTION
**Pull request by IT-DONG**
…efault  when brew install on MacOS

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
We probably don't want to alter the configuration shipped by upstream. They are the best placed to determine what are the “reasonable” defaults for users, not us. Also, if the config file format or config details change in the future, this will break our build.

**Comment by aaudiber** [comment]
@IT-DONG To avoid errors due to changes in `alluxio-site.properties.template`, can we update the recipe to create a new `alluxio-site.properties` file with just "alluxio.master.hostname=localhost" as its content?

@fxcoudert `localhost` is a good default for brew since the typical use case is to experiment with Alluxio on a single machine. But it isn't a good default in general since Alluxio usually runs on multiple machines in real deployments.

**Comment by IT-DONG** [comment]
@aaudiber Update the code as suggested!

**Comment by IT-DONG** [comment]
Update the code as suggested! @aaudiber @fxcoudert PTAL

**Comment by apc999** [comment]
Thanks @IT-DONG  for the fix! 

@fxcoudert , I and @aaudiber are maintainers for https://github.com/Alluxio/alluxio/ . This patch by @IT-DONG aims to fix https://github.com/Alluxio/alluxio/issues/8549 . Without this fix, the brew-installed Alluxio service on a local MacOS can not start

**Comment by apc999** [comment]
@fxcoudert can you help look at this patch? thanks!

File: Formula/alluxio.rb [hunk 1]
```diff
@@ -11,13 +11,35 @@ def install
     libexec.install Dir["*"]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
+    rm_rf Dir["#{etc}/alluxio/*"]
+
     (etc/"alluxio").install libexec/"conf/alluxio-env.sh.template" => "alluxio-env.sh"
     ln_sf "#{etc}/alluxio/alluxio-env.sh", "#{libexec}/conf/alluxio-env.sh"
+    (etc/"alluxio/alluxio-site.properties").write <<~EOS
```
Comment (fxcoudert) [inline Formula/alluxio.rb]: The default configuration should be provided (and maintained) in the upstream project, and installed as part of the default installation method. Or, at the very least, provided as an example file (`alluxio-site.properties.example`) which we can then rename and install as configuration file.

But we do not want to maintain the content of the software's configuration. If you, as the upstream project, decide to ship a product that cannot run without configuration, then that's your choice.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/alluxio.rb (hunk 1 above)
Comment (aaudiber) [reply Formula/alluxio.rb]: @fxcoudert Can we do the same thing Zookeeper does? https://github.com/Homebrew/homebrew-core/blob/master/Formula/zookeeper.rb#L39

```rb
  def default_alluxio_conf
    <<~EOS
      alluxio.master.hostname=localhost
    EOS
  end

  def install
    ...
    defaults = etc/"alluxio/alluxio-site.properties"
    defaults.write(default_alluxio_conf) unless defaults.exist?
```

We just need to set this single property

Comment (fxcoudert) [reply Formula/alluxio.rb]: That would be slightly better.

Comment (aaudiber) [reply Formula/alluxio.rb]: @fxcoudert better enough that we can fix this formula for people who want to use it? We have no intention to add more than just the single line of required configuration, so it really doesn't make sense to have a dedicated file just to hold the default master hostname value for running on a single node.

Comment (fxcoudert) [reply Formula/alluxio.rb]: OK then.

**Comment by aaudiber** [comment]
@IT-DONG, please simplify the config file to a single line as discussed, and squash the PR down to a single commit, thanks

**Comment by IT-DONG** [comment]
@aaudiber @fxcoudert DONE as discussed.

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**